src/track.h

```diff
@@ -18,113 +19,63 @@
-  gnss_signal_t sid;           /**< Satellite signal being tracked. */
-  u32 sample_count;            /**< Total num samples channel has tracked for. */
-  u32 code_phase_early;        /**< Early code phase. */
-  aided_tl_state_t tl_state;   /**< Tracking loop filter state. */
-  double code_phase_rate;      /**< Code phase rate in chips/s. */
-  u32 code_phase_rate_fp;      /**< Code phase rate in NAP register units. */
-  u32 code_phase_rate_fp_prev; /**< Previous code phase rate in NAP register units. */
-  s64 carrier_phase;           /**< Carrier phase in NAP register units. */
-  s32 carrier_freq_fp;         /**< Carrier frequency in NAP register units. */
-  s32 carrier_freq_fp_prev;    /**< Previous carrier frequency in NAP register units. */
-  double carrier_freq;         /**< Carrier frequency Hz. */
-  u32 corr_sample_count;       /**< Number of samples in correlation period. */
-  corr_t cs[3];                /**< EPL correlation results in correlation period. */
-  bit_sync_t bit_sync;         /**< Bit sync state. */
-  s8 bit_polarity;             /**< Polarity of nav message bits. */
-  u16 lock_counter;            /**< Lock counter. Increments when tracking new signal. */
-  cn0_est_state_t cn0_est;     /**< C/N0 Estimator. */
-  float cn0;                   /**< Current estimate of C/N0. */
-  u8 int_ms;                   /**< Integration length. */
-  u8 next_int_ms;              /**< Integration length for the next cycle. */
-  bool short_cycle;            /**< Set to true when a short 1ms integration is requested. */
-  bool output_iq;              /**< Set if this channel should output I/Q samples on SBP. */
-  u8 stage;                    /**< 0 = First-stage. 1 ms integration.
-                                    1 = Second-stage. After nav bit sync,
-                                    retune loop filters and typically (but
-                                    not necessarily) use longer integration. */
-  s8 elevation;                /**< Elevation angle, degrees */
-  alias_detect_t alias_detect; /**< Alias lock detector. */
-  lock_detect_t lock_detect;   /**< Phase-lock detector state. */
-  nav_bit_fifo_t nav_bit_fifo; /**< FIFO for navigation message bits. */
-  nav_time_sync_t nav_time_sync;  /**< Used to sync time decoded from navigation
-                                       message back to tracking channel. */
-} tracking_channel_t;
+
+typedef u8 tracker_channel_id_t;
 
 /** \} */
 
-/* Assuming we will never have a greater number of tracking channels than 12
- * We have to declare the number here as the number of tracking channels in
- * the FPGA is read at runtime. */
-/* TODO: NAP_MAX_N_TRACK_CHANNELS is a duplicate of MAX_CHANNELS */
-extern tracking_channel_t tracking_channel[NAP_MAX_N_TRACK_CHANNELS];
+void track_setup(void);
 
-void initialize_lock_counters(void);
+void tracking_send_state(void);
 
 float propagate_code_phase(float code_phase, float carrier_freq, u32 n_samples);
```

Comment:
this function uses following, which is GPS_L1 specific:
```
  /* Calculate the code phase rate with carrier aiding. */
  u32 code_phase_rate = (1.0 + carrier_freq/GPS_L1_HZ) *
                            NAP_TRACK_NOMINAL_CODE_PHASE_RATE;
```
I remember we agreed I will replace the magic numbers like GPS_L1_HZ with a function that takes sid as parameter. The question is how to do this in this function? There is no any SID passed to it.